commands/dice.js

```diff
@@ -2,37 +2,133 @@ var Roll = require('roll');
+              break;
+            }
+
+            if(k === (temp.length - 1)){
+              modifiers.push({value: temp.substring(j+1), sign: temp.charAt(j)});
+            }
+          }
+        }
+      }
+
+      //determine what kind of roll we are making
+      if(advantage != "none") {
+        var temp = (count > 0)? dicebag[i].replace(dicebag[i].charAt(0), "2"): "2"+dicebag[i];
+        var userRoll = roll.roll(temp);
+        var diceRolled = userRoll.rolled;
+        var result;
+
+        if(advantage === "advantage") {
+          result = (diceRolled[0] > diceRolled[1])? diceRolled[0]: diceRolled[1];
+        } else {
+          result = (diceRolled[0] > diceRolled[1])? diceRolled[1]: diceRolled[0];
+        }
+
+        if(modifiers.length > 0) {
+          for(var j = 0; j < modifiers.length; j++) {
+            if(modifiers[j].sign === "+") {
+              result += Number(modifiers[j].value);
+            } else {
+              result -= Number(modifiers[j].value);
+            }
           }
         }
-        //all the stats have been assembled, time to cast the die
+
+        msg.channel.send("You rolled "+dicebag[i]+" with "+advantage+".  Here are your rolls: "+diceRolled+".  With modifiers, your result is "+result);
+      } else {
         var userRoll = roll.roll(dicebag[i]);
         var diceRolled = userRoll.rolled;
+        var result = userRoll.result;
+        output = "You rolled "+dicebag[i]+".  ";
 
-        msg.channel.send("You rolled "+dicebag[i]+" and got the following: "+diceRolled+".\nAccounting for modifiers, your total is "+userRoll.result+".");
+        if(count > 1) {
+          output += "Here are your rolls: "+diceRolled+".  ";
+          if(modifiers.length > 0) {
+            output += "With modifiers, your result is "+result;
+          } else {
+            output += "Your result is "+result;
+          }
+        } else {
+          if(modifiers.length > 0) {
+            var original = result;
+            for (var j = 0; j < modifiers.length; j++) {
+              if(modifiers[j].sign === "+") {
+                original -= Number(modifiers[j].value);
+              } else {
+                original += Number(modifiers[j].value);
+              }
+            }
+            output += "Here is your roll: "+original+".  With modifiers, your result is "+result;
+          } else {
+            output += "Your result is: "+result;
+          }
+        }
+        msg.channel.send(output);
       }
+    } else {
+      output = "Bro, "+dicebag[i]+" is NOT in the proper format.";
+      msg.channel.send(output);
+      return;
     }
-    //userRoll = roll.roll(input);
-  } else {
-    msg.channel.send("That dice command is not in the proper format.");
-    return;
   }
+}
 
-  //msg.channel.send("Here are the dice you rolled: ["+userRoll.rolled+"]\nAnd here is the sum of those rolls: "+userRoll.result);
+function assembleDice(input) {
```

Comment:
This isn't necessarily a bad thing but it can be confusing. 
If you were to do a string like 'd6+1+d20', under normal order of operations that +1 could be applied to either the d6 or d20 but we are _always_ applying it to the right side with the way we split and reassemble the dicebag. 
It's also possible that things like 1+d6 the '1+' will just completely ignored.